python/GafferUI/ColorChooser.py

```diff
@@ -389,49 +405,174 @@ def __positionToColor( self, position ) :
+				return imath.V2f( position.x, self.__constraintPosition.y )
+			# Constrain to vertical line
+			return imath.V2f( self.__constraintPosition.x, position.y )
+
+		center = imath.V2f( self.bound().size() ) * 0.5
+		distanceToCenter = ( position - center ).length()
+
+		polarConstraintPosition = self.__cartesianToPolar( self.__constraintPosition, center, center.x )
+
+		# Extend line segment past the widget for distance check
+		extended = self.__polarToCartesian( imath.V2f( polarConstraintPosition.x, 2.0 ), center, center.x )
+
+		radiusLine = imath.Line3f( imath.V3f( center.x, center.y, 0 ), imath.V3f( extended.x, extended.y, 0 ) )
+		position3 = imath.V3f( position.x, position.y, 0 )
+		closestPointOnRadiusLine = radiusLine.closestPointTo( position3 )
+		distanceToRadiusLine = ( closestPointOnRadiusLine - position3 ).length()
+
+		constraintRadius = polarConstraintPosition.y * center.x
+		distanceToCircle = abs( distanceToCenter - constraintRadius )
+
+		dot = ( position - center ).dot( self.__constraintPosition - center )
+
+		polarPosition = self.__cartesianToPolar( position, center, center.x )
+
+		if self.__DragConstraints.X in self.__dragConstraints :
+			if (
+				abs( distanceToCircle ) < distanceToRadiusLine or
+				( dot < 0 and distanceToCenter > constraintRadius * 0.5 )
+			) :
+				# If we're closer to the circle than the radius line, or we're behind the center
+				# (from the point of view of `__constraintPosition`) and reasonably far from the center,
+				# constrain to the circle.
+				return self.__polarToCartesian(
+					imath.V2f( polarPosition.x, polarConstraintPosition.y ), center, center.x
+				)
+
+		if dot < 0 :
+			# If we're not constrained to the circle, but "behind" the center, snap to the center.
+			return center
+
+		# If no constraints are enforced yet, constrain to the radius line.
+		polarClosestPoint = self.__cartesianToPolar(
+			imath.V2f( closestPointOnRadiusLine.x, closestPointOnRadiusLine.y ), center, center.x
+		)
+		return self.__polarToCartesian(
+			imath.V2f( polarConstraintPosition.x, polarClosestPoint.y ),
+			center,
+			center.x
+		)
+
+	def __setupConstraints( self, event ) :
+
+		c, staticComponent = self.getColor()
+
+		self.__constraintPosition = self.__colorToPosition( c )
+
+		center = imath.V2f( self.bound().size() ) * 0.5
+
+		# Set `__dragConstraints` with `__constraintPosition` based on actual color
+		self.__dragConstraints = None
+		if event.modifiers == GafferUI.ButtonEvent.Modifiers.Control :
+			if not self.__useWheel() :
+				self.__dragConstraints = self.__DragConstraints.X | self.__DragConstraints.Y
+			else :
+				self.__dragConstraints = self.__DragConstraints.Y
+
+				center = imath.V2f( self.bound().size() ) * 0.5
+				if ( self.__constraintPosition - center ).length() > self.__snapRadius :
+					self.__dragConstraints |= self.__DragConstraints.X
+				else :
+					# Adjust `__constraintPosition` to be along the hue angle, slighly off center. This
```

Comment:
slighly -> slightly

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/6128/commits/59bde173999b22348195a2b7c79047888c27adc4